src/main/java/com/mathworks/ci/RunMatlabCommandBuilder.java

```diff
@@ -112,22 +116,40 @@ public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace,
     private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher,
             TaskListener listener, EnvVars envVars) throws IOException, InterruptedException {
         final String uniqueTmpFldrName = getUniqueNameForRunnerFile();
-        ProcStarter matlabLauncher;
+        final String uniqueCommandFile =
+                "command_" + getUniqueNameForRunnerFile().replaceAll("-", "_");
+
+        // Get unique temporary folder filepath
+        final FilePath uniqeTmpFolderPath =
+                getFilePathForUniqueFolder(launcher, uniqueTmpFldrName, workspace);
+        // Create a new command runner script in the temp folder.
+        final FilePath matlabCommandFile =
+                new FilePath(uniqeTmpFolderPath, uniqueCommandFile + ".m");
+        final String matlabCommandFileContent =
+                "cd '" + workspace.getRemote().replaceAll("'", "''") + "';\n" + getCommand();
+
+        // Display the commands on console output for users reference
+        listener.getLogger()
+                .println("Generating MATLAB script with content:\n" + getCommand() + "\n");
+
+        matlabCommandFile.write(matlabCommandFileContent, "UTF-8");

```

Comment:
MATLAB supports UTF-8 since 20a. Versions prior to 20a may not work correctly if we have non-ASCII characters in the command. We may have to document limitations regarding non-ASCII characters.